starlite/contrib/sqlalchemy/repository.py

```diff
@@ -193,7 +193,7 @@ async def get(self, item_id: Any, **kwargs: Any) -> ModelT:
             NotFoundError: If no instance found identified by `item_id`.
         """
         with wrap_sqlalchemy_exception():
-            statement = self._base_select(**kwargs)
+            statement = kwargs.pop("base_select", self.statement)
```

Comment:
Have you got an example of where you use this functionality? I.e., where you are managing the select statements outside of the repository and using the constructor `base_select` arg, or modifying `self.statement` on the repo is insufficient?

Reply:
yep, I thought you might ask this.  I'll put something together.

Reply:
I pinged you with a very specific use case.  But in short, it's for using the `query_expression` and `mapping a class against arbitrary subqueries`

https://docs.sqlalchemy.org/en/20/orm/nonstandard_mappings.html#mapping-a-class-against-arbitrary-subqueries
And this: https://docs.sqlalchemy.org/en/20/orm/queryguide/columns.html#using-with-expression-with-unions-other-subqueries